Lib/test/test_capi/test_number.py

```diff
@@ -0,0 +1,612 @@
+        # Test PyNumber_divmod()
+        divmod_ = _testcapi.number_divmod
+
+        self.assertEqual(divmod_(4, 3), divmod(4, 3))
+
+        # CRASHES divmod_(NULL, 42)
+        # CRASHES divmod_(42, NULL)
+
+    def test_power(self):
+        # Test PyNumber_Power()
+        power = _testcapi.number_power
+
+        self.assertEqual(power(4, 3), pow(4, 3))
+        self.assertEqual(power(0.5, 2), pow(0.5, 2))
+        self.assertEqual(power(2, -1.0), pow(2, -1.0))
+        self.assertEqual(power(4, 11, 5), pow(4, 11, 5))
+        self.assertEqual(power(4, IntSubclass(11)), pow(4, 11))
+        self.assertEqual(power(4, IntSubclass2(11)), 11)
+        self.assertEqual(power(4, IntSubclass2(NotImplemented), NULL), 1)
+
+        self.assertRaises(TypeError, power, "spam", 42)
+        self.assertRaises(TypeError, power, object(), 42)
+        self.assertRaises(TypeError, power, 42, "spam")
+        self.assertRaises(TypeError, power, 42, object())
+        self.assertRaises(TypeError, power, 1, 2, "spam")
+        self.assertRaises(TypeError, power, 1, 2, object())
+
+        # CRASHES power(NULL, 42)
+        # CRASHES power(42, NULL)
+
+    def test_negative(self):
+        # Test PyNumber_Negative()
+        negative = _testcapi.number_negative
+
+        self.assertEqual(negative(42), -42)
+
+        self.assertRaises(TypeError, negative, BadAdd())
+        self.assertRaises(TypeError, negative, object())
+        self.assertRaises(SystemError, negative, NULL)
+
+    def test_positive(self):
+        # Test PyNumber_Positive()
+        positive = _testcapi.number_positive
+
+        self.assertEqual(positive(-1), +(-1))
+
+        self.assertRaises(TypeError, positive, BadAdd())
+        self.assertRaises(TypeError, positive, object())
+        self.assertRaises(SystemError, positive, NULL)
+
+    def test_absolute(self):
+        # Test PyNumber_Absolute()
+        absolute = _testcapi.number_absolute
+
+        self.assertEqual(absolute(-1), abs(-1))
+
+        self.assertRaises(TypeError, absolute, BadAdd())
+        self.assertRaises(TypeError, absolute, object())
+        self.assertRaises(SystemError, absolute, NULL)
+
+    def test_invert(self):
+        # Test PyNumber_Invert()
+        invert = _testcapi.number_invert
+
+        self.assertEqual(invert(123), ~123)
+
+        self.assertRaises(TypeError, invert, BadAdd())
+        self.assertRaises(TypeError, invert, object())
+        self.assertRaises(SystemError, invert, NULL)
+
+    def test_lshift(self):
+        # Test PyNumber_Lshift()
+        lshift = _testcapi.number_lshift
+
+        self.assertEqual(lshift(3, 5), 3 << 5)
+
+        # CRASHES lshift(NULL, 1)
+        # CRASHES lshift(1, NULL)
+
+    def test_rshift(self):
+        # Test PyNumber_Rshift()
+        rshift = _testcapi.number_rshift
+
+        self.assertEqual(rshift(5, 3), 5 >> 3)
+
+        # RHS implements __rrshift__ but returns NotImplemented
+        with self.assertRaises(TypeError) as context:
```

Comment:
All this have no relation to the C API.

Reply:
Not sure I get your point.  This (and cases below) are related to
https://github.com/python/cpython/blob/18203a6bc9ccf3a9ba901574dfc772474b027e2a/Objects/abstract.c#L989-L1006

This is a part of C API.  It was tested in test_print.py (partially), but I think we should have some variants here as well.

Reply:
Then call it directly: `rshift(print, 42)`.

Although it is a CPython implementation detail, so it it is better to move it in a separate test method and decorate as cpython_only.

Reply:
Then test_stream_redirection_hint_for_py2_migration() should be decorated as well.